src/style/text_style.rs

```diff
@@ -12,20 +12,18 @@ pub trait TextStyle {
     /// Color type.
     type Color: PixelColor;
 
-    /// Render a text object using this style.
-    fn render_text<D>(&self, text: &Text<'_>, target: &mut D) -> Result<(), D::Error>
+    /// Renders a single line of text using this style.
+    ///
+    /// Returns the offset from the current position to the start of the next line.
+    fn render_text<D>(
+        &self,
+        text: &str,
+        position: Point,
+        target: &mut D,
+    ) -> Result<Point, D::Error>
     where
         D: DrawTarget<Color = Self::Color>;
 
-    /// Returns the bounding box of a text object rendered using this style.
-    fn bounding_box(&self, text: &Text<'_>) -> Rectangle;
-}
-
-/// Pixels iterator.
-pub trait TextStylePixels<'a>: TextStyle {
-    /// Iterator type.
-    type Iter: Iterator<Item = Pixel<Self::Color>> + 'a;
-
-    /// Returns an iterator over the drawn pixels.
-    fn pixels(&self, text: &Text<'a>) -> Self::Iter;
+    /// Returns the bounding box of a single line of text rendered using this style.
+    fn bounding_box(&self, text: &str, position: Point) -> (Rectangle, Point);
```

Comment:
Should we perhaps name this `first_line_bounding_box` or something like that to be more specific? If I didn't read the docs, I'd assume this `bounding_box` method would behave like all the others in e-g and give me the bounding box of the complete, potentially multiline, string.

I'm also not sure what the `Point` being returned does, so explaining that would help me and others.

Presumably an impl should only test the first line of any given text? Either way, defining this behaviour in another `# Implementation notes` section would help.

Reply:
> Should we perhaps name this `first_line_bounding_box` or something like that to be more specific?

I've renamed it to `line_bounding_box`.

> I'm also not sure what the `Point` being returned does, so explaining that would help me and others.

This might change in future PR and I've only added a TODO to remember to document this later.

> Presumably an impl should only test the first line of any given text?

It will only be called with single line strings.